tlatools/org.lamport.tlatools/src/util/SimpleFilenameToStream.java

```diff
@@ -65,66 +70,43 @@ public SimpleFilenameToStream(final String libraryPath) {
  * (which is being used in the default constructor).
  */
   public SimpleFilenameToStream(String[] anLibraryPaths) {
-	  libraryPaths = getLibraryPaths(getInstallationBasePath(), anLibraryPaths);
+    userFileResolver = getLibraryPaths(anLibraryPaths);
+    standardLibraryResolver = getInstallationBasePath();
   }
 
   /**
-   * Find the absolute path in the file system to the directory
-   * containing the TLA+ standard modules; this path
-   * must have separators appropriate to the Unix ('/') or Windows ('\') world.
+   * Find the TLA+ standard modules.
    *
-   * @return the location of the TLA+ standard modules as an absolute path or as a "jar:" URL
+   * @return a resolver for locating the TLA+ standard modules
    */
-  private String getInstallationBasePath() {
-
-    // get a "file:" URL for the base directory for package tla2sany
-    final URL         url = cl.getResource(STANDARD_MODULES);
-
-    // jar expanded to the fs (make sure to handle whitespaces correctly
-    final String path = url.toString();
-	try {
-    	// convert to URI which handles paths correctly (even OS dependently)
-    	if(!FilenameToStream.isInJar(path)) {
-    	final URI uri = new URI(path);
-    		return new File(uri).getAbsolutePath();
-    	}
-    } catch (URISyntaxException e) {
-    	System.err.println(path);
-    	// may never happen
-    	e.printStackTrace();
-    } catch (IllegalArgumentException e) {
-    	System.err.println(path);
-    	// may never happen
-    	e.printStackTrace();
-    }
-    return path;
-   }
+  private static Resolver getInstallationBasePath() {
+    return new ClasspathResolver(cl, STANDARD_MODULES);
```

Comment:
If this is the only consumer of `STANDARD_MODULES`, consider renaming it to `STANDARD_MODULES_INTERNAL_PATH` or similar.

Reply:
This is a fine suggestion, but I think I'm going to defer it. The diff in this PR is big enough already. 😓